src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala

```diff
@@ -48,39 +48,39 @@ trait WatchableTests[F[_], Resource <: { def metadata: Option[ObjectMeta] }]
       _ = status shouldBe Status.Ok
     } yield ()
 
-    val expected: Set[EventType] = Set(EventType.ADDED, EventType.MODIFIED, EventType.DELETED)
+    val expected = Set[EventType](EventType.ADDED, EventType.MODIFIED, EventType.DELETED)
 
     def processEvent(
         received: Ref[F, mutable.Set[EventType]],
         signal: SignallingRef[F, Boolean]
     ): Pipe[F, Either[String, WatchEvent[Resource]], Unit] =
       _.flatMap {
         case Right(we) if we.`object`.metadata.exists(_.name.exists(_ == name)) =>
-          fs2.Stream.eval {
+          Stream.eval {
             for {
               _           <- received.update(_ += we.`type`)
               allReceived <- received.get.map(_.intersect(expected) == expected)
               _           <- F.whenA(allReceived)(signal.set(true))
             } yield ()
           }
-        case _ => fs2.Stream.eval(Applicative[F].unit)
+        case _ => Stream.eval(Applicative[F].unit)
       }
 
     val watchStream = for {
-      signal   <- fs2.Stream.eval(SignallingRef[F, Boolean](false))
-      received <- fs2.Stream.eval(Ref.of(mutable.Set.empty[EventType]))
+      signal   <- Stream.eval(SignallingRef[F, Boolean](false))
+      received <- Stream.eval(Ref.of(mutable.Set.empty[EventType]))
       watch <- watchApi(namespaceName).watch
         .through(processEvent(received, signal))
         .evalMap(_ => received.get)
         .interruptWhen(signal)
-        .concurrently(fs2.Stream.eval(events))
     } yield watch
 
-    val timeout = 60.seconds
-    val actual  = watchStream.interruptAfter(timeout).compile.toList
-    actual.map { set =>
-      val a = set.headOption.getOrElse(fail("stream should have at least one element with all received events"))
-      a should === (expected)
-    }
+    val watchEvents = for {
+      set <- watchStream.interruptAfter(60.seconds).compile.toList
+      result = set.headOption.getOrElse(fail("stream should have at least one element with all received events"))
+      _      = result should === (expected)
+    } yield ()
+
+    List(watchEvents, timer.sleep(1.second) *> sendEvents).parSequence
```

Comment:
Is this 1 second delay really needed here?

Reply:
That's what made it pass. The events where sent ahead of the watch start.

Reply:
Interesting. I thought etcd always sends all events occured within the last 5 minutes.